### PR TITLE
Movies dialogs

### PR DESCRIPTION
**Pull request by jasonkliu**
Changes:

* Moved `Videos/JSONReader.java` to `JSONReader.java` and updated context
* Use Activity instead of Context in `Cache/CalendarCache.java`

### CODE REVIEW

File: src/edu/yalestc/yalepublic/Cache/CalendarCache.java [hunk 1]
```diff
@@ -168,9 +169,13 @@ private void createDatabaseForTheFirstTime(int year, int month){
             Log.i("CacheUpdater","Created puller with query" + queries[i]);
              Log.i("CacheUpdater", "yearMonth " + yearMonth[i]);
             if (results[i] == null) {
-                Toast toast = new Toast(mContext);
-                toast = Toast.makeText(mContext, "You need internet connection to succesfully finish", Toast.LENGTH_LONG);
-                toast.show();
+                mActivity.runOnUiThread(new Runnable() {
```
Comment (jasonkliu) [inline src/edu/yalestc/yalepublic/Cache/CalendarCache.java]: Is the purpose of changing Context to Activity to more easily display this toast?

File: src/edu/yalestc/yalepublic/Events/EventsJSONReader.java [hunk 2]
```diff
@@ -65,10 +63,14 @@ protected void onPreExecute(){
     protected String doInBackground(Void... voids) {
         mRawData = super.getData();
         if (mRawData == null) {
-            Toast toast = new Toast(mContext);
-            toast = Toast.makeText(mContext, "You need internet connection to view the content!", Toast.LENGTH_LONG);
-            toast.show();
-            Log.i("CalendarFragment", "Failure");
+            mActivity.runOnUiThread(new Runnable() {
```
Comment (jasonkliu) [inline src/edu/yalestc/yalepublic/Events/EventsJSONReader.java]: Same question here as above.

File: src/edu/yalestc/yalepublic/Cache/CalendarCache.java (hunk 1 above)
Comment (Ebstein) [inline src/edu/yalestc/yalepublic/Cache/CalendarCache.java]: To display it at all :)

Comment (jasonkliu) [inline src/edu/yalestc/yalepublic/Cache/CalendarCache.java]: Did the previous `Toast toast = new Toast(mContext);` not work? It seems that we've used that paradigm for a while; or the alternate `Toast toast = new Toast(getApplicationContext());`

**Pull request merged**